packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -1,125 +1,139 @@
+
+	// Must have at least one text color.
+	if ( ! currentTextColor && ! currentLinkColor ) {
+		return null;
+	}
+
+	const textColors = [
+		{
+			color: currentTextColor,
+			description: __( 'text color' ),
+		},
+		{
+			color: currentLinkColor,
+			description: __( 'link color' ),
+		},
+	];
+	const colordBackgroundColor = colord( currentBackgroundColor );
 	const backgroundColorHasTransparency = colordBackgroundColor.alpha() < 1;
-	const hasTransparency =
-		textColorHasTransparency || backgroundColorHasTransparency;
-	const isReadable = colordTextColor.isReadable( colordBackgroundColor, {
+	const backgroundColorBrightness = colordBackgroundColor.brightness();
+	const isReadableOptions = {
 		level: 'AA',
 		size:
 			isLargeText || ( isLargeText !== false && fontSize >= 24 )
 				? 'large'
 				: 'small',
-	} );
+	};
 
-	// Don't show the message if the text is readable AND there's no transparency.
-	// This is the default.
-	if ( isReadable && ! hasTransparency ) {
-		return null;
-	}
+	let message = '';
+	let speakMessage = '';
+	for ( const item of textColors ) {
+		// If there is no color, go no further.
+		if ( ! item.color ) {
+			continue;
+		}
+		const colordTextColor = colord( item.color );
+		const isColordTextReadable = colordTextColor.isReadable(
+			colordBackgroundColor,
+			isReadableOptions
+		);
+		const textHasTransparency = colordTextColor.alpha() < 1;
 
-	if ( hasTransparency ) {
-		if (
-			// If there's transparency, don't show the message if the alpha checker is disabled.
-			! enableAlphaChecker ||
-			// If the alpha checker is enabled, we only show the warning if the text has transparency.
-			( isReadable && ! textColorHasTransparency )
-		) {
-			return null;
+		// If the contrast is not readable.
+		if ( ! isColordTextReadable ) {
+			// Don't show the message if the background or text is transparent.
+			if ( backgroundColorHasTransparency || textHasTransparency ) {
+				continue;
+			}
+			message =
+				backgroundColorBrightness < colordTextColor.brightness()
+					? sprintf(
+							// translators: %s is a type of text color, e.g., "text color" or "link color"
+							__(
+								'This color combination may be hard for people to read. Try using a darker background color and/or a brighter %s.'
+							),
+							item.description
+					  )
+					: sprintf(
+							// translators: %s is a type of text color, e.g., "text color" or "link color"
+							__(
+								'This color combination may be hard for people to read. Try using a brighter background color and/or a darker %s.'
+							),
+							item.description
+					  );
+			speakMessage = __(
+				'This color combination may be hard for people to read.'
+			);
+			// Break from the loop when we have a contrast warning.
+			// These messages take priority over the transparency warning.
+			break;
+		}
+
+		// If the text color is readable, but transparent, show the transparent warning.
+		if ( textHasTransparency && true === enableAlphaChecker ) {
```

Comment:
```suggestion
		if ( textHasTransparency && enableAlphaChecker ) {
```

Don't think we need the yoda condition here.